src/services/environment-manager.js

```diff
@@ -68,13 +72,47 @@ function EnvironmentManager(config) {
       to: toEnvironmentId,
     };
 
-    return agent
+    const deploymentRequestResponse = await agent
       .post(`${config.serverHost}/api/deployment-requests`)
       .set('Authorization', `Bearer ${authToken}`)
       .set('forest-origin', 'Lumber')
       .set('forest-project-id', config.projectId)
-      .send(DeploymentRequestSerializer.serialize(deploymentRequest))
-      .then(() => true);
+      .send(DeploymentRequestSerializer.serialize(deploymentRequest));
+
+    if (!deploymentRequestResponse
+        || !deploymentRequestResponse.body
+        || !deploymentRequestResponse.body.meta
+        || !deploymentRequestResponse.body.meta.job_id) {
+      return false;
+    }
+
+    const jobId = deploymentRequestResponse.body.meta.job_id;
+
+    const bar = new ProgressBar('Copying layout [:bar] :percent', { total: 100 });
+
+    const checkState = async function checkState() {
+      const jobResponse = await agent
+        .get(`${config.serverHost}/api/jobs/${jobId}`)
+        .set('Authorization', `Bearer ${authToken}`)
+        .set('forest-origin', 'Lumber')
+        .set('forest-project-id', config.projectId);
+
+      if (jobResponse
+          && jobResponse.body
+          && jobResponse.body.state
+          && jobResponse.body.progress) {
+        bar.tick(jobResponse.body.progress);
+        if (jobResponse.body.state !== 'inactive'
```

Comment:
No ! this is supposed to work even if the progress is not present or you will do an infinite loop

Reply:
You need to move this if outside of the other if